File: CS/T119350/fmDesigner.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using System.ComponentModel.Design;
using DevExpress.DataAccess.Sql;
using DevExpress.XtraReports.UserDesigner;
using DevExpress.DataAccess.Native;
using DevExpress.DataAccess.Wizard.Services;

namespace T119350 {
    public partial class fmDesigner : Form {
        CustomConnectionStorageService connectionStorageService;

        public fmDesigner() {
            InitializeComponent();

            connectionStorageService = new CustomConnectionStorageService() {
                FileName = "connections.xml",
                IncludeApplicationConnections = false
            };
            ReplaceService(this.reportDesignerMDIController, typeof(IConnectionStorageService), connectionStorageService);
            reportDesignerMDIController.DesignPanelLoaded += DesignMdiControllerOnDesignPanelLoaded;
        }

        private void ReplaceService(IServiceContainer container, Type serviceType, object serviceInstance) {
            if (container.GetService(serviceType) != null)
                container.RemoveService(serviceType);
            container.AddService(serviceType, serviceInstance);
        }

        private void DesignMdiControllerOnDesignPanelLoaded(object sender, DesignerLoadedEventArgs e) {
            ReplaceService(e.DesignerHost, typeof(IConnectionStorageService), connectionStorageService); // insert this line!

            IComponentChangeService componentChangeService =
            (IComponentChangeService)e.DesignerHost.GetService(typeof(IComponentChangeService));
            if(componentChangeService != null) {
                componentChangeService.ComponentAdded += ComponentChangeServiceOnComponentAdded;
                componentChangeService.ComponentChanged += ComponentChangeServiceOnComponentChanged;
            }
        }

        private void ComponentChangeServiceOnComponentAdded(object sender, ComponentEventArgs e) {
            SqlDataSource sqlDataSource = e.Component as SqlDataSource;
            if (sqlDataSource == null)
                return;
            connectionStorageService.SaveConnection(sqlDataSource.Connection);
        }

        private void ComponentChangeServiceOnComponentChanged(object sender, ComponentChangedEventArgs e) {
            SqlDataSource sqlDataSource = e.Component as SqlDataSource;
            if (sqlDataSource == null)
                return;
            connectionStorageService.SaveConnection(sqlDataSource.Connection);
        }

        private void fmDesigner_Load(object sender, EventArgs e) {
            //Open new report
            XtraReport report = new XtraReport();
            this.reportDesignerMDIController.OpenReport(report);
        }
    }
}

File: VB/T119350/fmDesigner.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports System.ComponentModel.Design
Imports DevExpress.DataAccess.Sql
Imports DevExpress.XtraReports.UserDesigner
Imports DevExpress.DataAccess.Wizard.Services

Namespace T119350

    Public Partial Class fmDesigner
        Inherits Form

        Private connectionStorageService As CustomConnectionStorageService

        Public Sub New()
            InitializeComponent()
            connectionStorageService = New CustomConnectionStorageService() With {.FileName = "connections.xml", .IncludeApplicationConnections = False}
            ReplaceService(reportDesignerMDIController, GetType(IConnectionStorageService), connectionStorageService)
            AddHandler reportDesignerMDIController.DesignPanelLoaded, AddressOf DesignMdiControllerOnDesignPanelLoaded
        End Sub

        Private Sub ReplaceService(ByVal container As IServiceContainer, ByVal serviceType As Type, ByVal serviceInstance As Object)
            If container.GetService(serviceType) IsNot Nothing Then container.RemoveService(serviceType)
            container.AddService(serviceType, serviceInstance)
        End Sub

        Private Sub DesignMdiControllerOnDesignPanelLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
            ReplaceService(e.DesignerHost, GetType(IConnectionStorageService), connectionStorageService) ' insert this line!
            Dim componentChangeService As IComponentChangeService = CType(e.DesignerHost.GetService(GetType(IComponentChangeService)), IComponentChangeService)
            If componentChangeService IsNot Nothing Then
                AddHandler componentChangeService.ComponentAdded, AddressOf ComponentChangeServiceOnComponentAdded
                AddHandler componentChangeService.ComponentChanged, AddressOf ComponentChangeServiceOnComponentChanged
            End If
        End Sub

        Private Sub ComponentChangeServiceOnComponentAdded(ByVal sender As Object, ByVal e As ComponentEventArgs)
            Dim sqlDataSource As SqlDataSource = TryCast(e.Component, SqlDataSource)
            If sqlDataSource Is Nothing Then Return
            connectionStorageService.SaveConnection(sqlDataSource.Connection)
        End Sub

        Private Sub ComponentChangeServiceOnComponentChanged(ByVal sender As Object, ByVal e As ComponentChangedEventArgs)
            Dim sqlDataSource As SqlDataSource = TryCast(e.Component, SqlDataSource)
            If sqlDataSource Is Nothing Then Return
            connectionStorageService.SaveConnection(sqlDataSource.Connection)
        End Sub

        Private Sub fmDesigner_Load(ByVal sender As Object, ByVal e As EventArgs)
            'Open new report
            Dim report As XtraReport = New XtraReport()
            reportDesignerMDIController.OpenReport(report)
        End Sub
    End Class
End Namespace
